### PR TITLE
Add draft for GREP: Remove msgq API

### PR DESCRIPTION
**Pull request by mbr0wn**

### CODE REVIEW

File: grep-00XX-remove-msgq.md [hunk 1]
```diff
@@ -0,0 +1,62 @@
...
+
+The following files shall be removed:
+
+- Everything in `gnuradio/messages` and `gnuradio-runtime/lib/messages`, as well
+  as all related SWIG bindings
+- `msg_queue.{cc,h}` and related SWIG bindings and unit tests
+- gr-blocks: `bin_statistics` block.
+  - Alternative: Replace this block with one that uses actual message passing
+- gr-digital: `framer_sink` and `packet_sink` blocks
+  - Alternative: Replace these blocks with ones that uses actual message passing
```
Comment (dkozel) [inline grep-00XX-remove-msgq.md]: Minor:
ones that _use_ actual message passing

File: grep-00XX-remove-msgq.md [hunk 2]
```diff
@@ -0,0 +1,62 @@
...
+streams to messages. In the past, this API was used to move data from the C++
+into the Python domain, because the msgq API calls could be easily SWIG'd, but
+the block connections could not.
+
+For a few years now, msgq has been superseded by Python blocks and message
+passing. The former solves the issue moving data from the C++ to the Python
+domain, and the latter implements a more feature-complete message passing API.
+
+The msgq API is thus obsolete, and can be easily replaced with those techniques.
+For the upcoming 3.8 release, it is already clear that the msgq API will require
```
Comment (noc0lour) [inline grep-00XX-remove-msgq.md:32]: Hm, 3.8 has been already released, is msgq still in there?

**Review by noc0lour — APPROVED** [review]
Less cruft we cannot maintain :+1:

**Comment by mbr0wn** [comment]
This is another master-branch-level decision, thus goes to @mormj.

**Comment by mormj** [comment]
Closing since msgq API got deleted along the way

**Pull request closed without merging**